ibis/expr/types.py

```diff
@@ -1239,6 +1243,38 @@ def _implicit_cast(self, arg):
         return TimestampScalar(op)
 
 
+class IntervalValue(IntegerValue):
+
+    def __init__(self, meta=None):
+        self.meta = meta
+        self._unit = getattr(meta, 'unit', 's')
+
+    def type(self):
+        return dt.Interval(unit=self._unit)
+
+    # @classmethod
+    # def _make_constructor(cls, meta):
+    #     def constructor(arg, name=None):
+    #         return cls(arg, meta, name=name)
+    #     return constructor
+
+    def _can_compare(self, other):
+        return isinstance(other, IntervalValue)
+
+    def _can_implicit_cast(self, arg):
+        print('can_implicit_cast')
+        op = arg.op()
+        if isinstance(op, Literal):
+            if isinstance(op.value, six.integer_types):
```

Comment:
Arbitrary integers shouldn't be implicitly castable to `IntervalValue`s since integers do not have units. Only other interval values should be implicitly castable to interval values.